### PR TITLE
`:completion vertico` improvements

### PR DESCRIPTION
**Pull request by iyefrat**
- minor code reorganization
- which-key embark indicator improvements
- `+vertico/search-symbol-at-point` preview fix
- bump packages
- add [consult-dir](https://github.com/karthink/consult-dir)

### CODE REVIEW

**Comment by minad** [comment]
@iyefrat There has been a change in vertico-repeat. The package does not activate itself anymore automatically:

https://github.com/minad/vertico/commit/ff63d353bef62a7c66834200a564a00d1a92c833#diff-42d2775dfb0ba97a32545981ad1df8d637721deb57d035f65bd5e56ed20b7725

**Comment by iyefrat** [comment]
@minad thanks. What was the reason for this change? Also, is `vertico-repeat` purposely not autoloaded now or is that a mistake?

**Comment by minad** [comment]
It is better style this way, since the package does not autoactivate itself by only installing it. 

I could also add an autoload to vertico-repeat, however the first time you run vertico-repeat without having started a vertico session before you will only get an error message.

**Comment by minad** [comment]
See https://github.com/minad/vertico/commit/665660f91004e4c6aad1e18b9cdc256607228c42

**Review by hlissner — CHANGES_REQUESTED** [review]

File: modules/completion/vertico/config.el [hunk 1]
```diff
@@ -155,6 +174,15 @@ overrides `completion-styles' during company completion sessions.")
...
   :config
   (set-popup-rule! "^\\*Embark Export Grep" :size 0.35 :ttl 0 :quit nil)
+
+  (defadvice! +vertico--embark-which-key-prompt-a (fn &rest args)
+    "Hide the which-key indicator immediately when using the completing-read prompter."
+    :around #'embark-completing-read-prompter
+    (when-let ((win (get-buffer-window which-key--buffer
+                                       'visible)))
+      (quit-window 'kill-buffer win)
+      (let ((embark-indicators (delq #'+vertico-embark-which-key-indicator embark-indicators)))
```
[the remark this answers is not in the document]
Comment (iyefrat) [reply modules/completion/vertico/config.el]: Done, there has also been a simplification of this function in the wiki, so I added that as well.

File: modules/completion/vertico/config.el [hunk 2]
```diff
@@ -134,6 +148,11 @@ overrides `completion-styles' during company completion sessions.")
         :desc "Select candidate" "TAB" #'+vertico/crm-select
         :desc "Enter candidates" "RET" #'+vertico/crm-exit))
 
+(use-package consult-dir
+  :bind (("C-x C-d" . consult-dir)
+         :map vertico-map
+         ("C-x C-d" . consult-dir)
+         ("C-x C-j" . consult-dir-jump-file)))
```
[the remark this answers is not in the document]
Comment (iyefrat) [reply modules/completion/vertico/config.el]: Done.

**Comment by iyefrat** [comment]
Also I pushed a formatting commit, a function was changed a while back but the formatting slipped by.

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
LGTM. Thanks again for your help!

**Pull request merged**